File: src/telemetry_dialyzer_problem.erl
-module(telemetry_dialyzer_problem).

-export([telemetry_problem/0, general_problem/0]).

telemetry_problem() ->
   otel_telemetry:start_telemetry_span(my_tracer, <<"my_event">>, #{}, #{}).

general_problem() ->
    Tracer = opentelemetry:get_application_tracer(?MODULE),
    otel_tracer:start_span(Tracer, <<"my_span">>, #{}).

File: lib/telemetry_dialyzer_problem.ex
defmodule TelemetryDialyzerProblem do
  @moduledoc """
  Documentation for `TelemetryDialyzerProblem`.
  """

  @doc """
  Hello world.

  ## Examples

      iex> TelemetryDialyzerProblem.hello()
      :world

  """
  def hello do
   OpentelemetryTelemetry.start_telemetry_span(:my_tracer, "my_event", %{}, %{})
  end
end
